Assets/SEE/UI/DebugAdapterProtocol/DebugAdapterProtocolSession.cs

```diff
@@ -0,0 +1,1233 @@
+            }
+            ConsoleWindow.AddMessage($"Start debugging session: {Adapter.Name} - {Adapter.AdapterFileName} - {Adapter.AdapterArguments} - {Adapter.AdapterWorkingDirectory}\n");
+
+            // starts the debug adapter process
+            if (!CreateAdapterProcess())
+            {
+                string message = "Couldn't create the debug adapter process.";
+                ConsoleWindow.AddMessage(message, "Adapter", "Error");
+                Debug.LogWarning(message);
+                Destroyer.Destroy(this);
+                return;
+            }
+            else
+            {
+                ConsoleWindow.AddMessage("Created the debug adapter process.\n", "Adapter", "Log");
+            }
+            // starts the debug adapter host
+            if (!CreateAdapterHost())
+            {
+                string message = "Couldn't create the debug adapter host.";
+                ConsoleWindow.AddMessage(message, "Adapter", "Error");
+                Debug.LogWarning(message);
+                Destroyer.Destroy(this);
+                return;
+            }
+            else
+            {
+                ConsoleWindow.AddMessage("Created the debug adapter host.\n", "Adapter", "Log");
+            }
+
+            // adds breakpoint events
+            DebugBreakpointManager.OnBreakpointAdded += OnBreakpointsChanged;
+            DebugBreakpointManager.OnBreakpointRemoved += OnBreakpointsChanged;
+
+            // sends the initialize request
+            try
+            {
+                capabilities = adapterHost.SendRequestSync(new InitializeRequest()
+                {
+                    PathFormat = InitializeArguments.PathFormatValue.Path,
+                    ClientID = "vscode",
+                    ClientName = "Visual Studio Code",
+                    AdapterID = Adapter.Name,
+                    Locale = "en",
+                    LinesStartAt1 = true,
+                    ColumnsStartAt1 = true,
+                });
+                if (capabilities.SupportsSteppingGranularity != true)
+                {
+                    steppingGranularity = null;
+                }
+            }
+            catch (Exception e)
+            {
+                ConsoleWindow.AddMessage(e.Message + "\n", "Adapter", "Error");
+                Debug.LogWarning(e);
+                Destroyer.Destroy(this);
+            }
+        }
+
+        /// <summary>
+        /// This component supports the desktop platform.
+        /// </summary>
+        protected override void StartDesktop()
+        {
+        }
+
+        /// <summary>
+        /// Executes the queued actions on the main thread.
+        /// Waits until the capabilities are known.
+        /// </summary>
+        private void Update()
+        {
+            if (capabilities != null)
+            {
+                while (actions.Count > 0)
+                {
+                    actions.Dequeue()();
+                }
+            }
```

Comment:
The `actions` are mainly responsible for sending and receiving data to and from the debug adapter. Doing that `async` isn't allowed.

(the documentation of the `update` methods already mentioned that, but I added it to `actions` as well)

Reply:
Hmm, I don't really understand yet. `async` (at least when used with `UniTask`) is still executed on the player loop (`await` usually just defers to the next frame), and hence shouldn't affect what's allowed or not. The only way that would change is if the task is executed on the thread pool, but that's not necessary here.

Reply:
I am still not quite sure what the actual reason is, but I tried using `async` first, but the `adapterHost.SendRequest` and `adapterHost.SendRequestSync` methods aren't allowed in there.

So I switched to the queue-solution, since (almost) all actions send data to the debug adapter with the 2 mentioned methods.

Reply:
I'm still confused why this is the case to be honest, but it's probably not worth the effort to investigate it more deeply right now (especially considering that the current implementation works just as well), so I'll resolve this for now.